File: LeetCode/[M] - 739. Daily Temperatures/solution.py
from typing import List


class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        lengthT = len(temperatures)
        stack = [lengthT - 1]
        res = [0] * lengthT

        for i in range(lengthT - 2, -1, -1):
            while stack and temperatures[i] >= temperatures[stack[-1]]:
                stack.pop()
            if not stack:
                res[i] = 0
            else:
                res[i] = stack[-1] - i
            stack.append(i)
        return res

File: LeetCode/[M] - 739. Daily Temperatures/solution.go
package main

func dailyTemperatures(temperatures []int) []int {
	stack := make([]int, 0, len(temperatures))
	stack = append(stack, len(temperatures)-1)
	res := make([]int, len(temperatures))

	for i := len(temperatures) - 2; i > -1; i-- {
		for len(stack) != 0 && temperatures[i] >= temperatures[stack[len(stack)-1]] {
			stack = stack[:len(stack)-1]
		}
		if len(stack) == 0 {
			res[i] = 0
		} else {
			res[i] = stack[len(stack)-1] - i
		}
		stack = append(stack, i)
	}
	return res
}
